### PR TITLE
Refactor of drive controller

### LINKED ISSUE #49: Make DriveController a component we only add to controllable cars
DriveController currently has a flag that says we should ignore the input if it is not controlled by the client. Would it not make sense to just only add this component to a car that needs to be controlled.

### PR DESCRIPTION
**Pull request by RileyEv**
All cars now have the `Car` mono behaviour by default.

When a car is set to controllable the `DriveController` component is added which allows the car to be controlled.

Removes the need for a flag in `DriveController` to detect if it should ignore user inputs.

Resolves #49

### CODE REVIEW

File: ProjectFiles/Assets/Scripts/Car/Car.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        public List<Axle> axles;
+        public CarProperties carProperties;
+        private DriveController driveController;
+
+
+        public void SetControllable()
+        {
+            // Add DriveController to car
+            driveController = gameObject.AddComponent(typeof(DriveController)) as DriveController;
```
Comment (Keywarn) [inline ProjectFiles/Assets/Scripts/Car/Car.cs]: `gameObject.AddCopmonent<DriveController>();`

File: ProjectFiles/Assets/Scripts/Car/Car.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            HUD hud = FindObjectOfType<HUD>();
+            hud.Car = GetComponent<Rigidbody>();
+        }
+    }
+    
+    // Stores axles
+    [Serializable]
+    public class Axle
+    {
+        [field: SerializeField] public WheelCollider LeftWheel { get; private set; }
```
Comment (Keywarn) [inline ProjectFiles/Assets/Scripts/Car/Car.cs]: Don't think this needs to be serialized if it's public

**Review by Keywarn — CHANGES_REQUESTED** [review]
Couple of things. Names could do with a bit of clarification.

**Comment by RileyEv** [comment]
Made some minor refactors.

- `Car` -> `CarFactory`
- Axles are now set by `CarFactory` rather than fetched from the `CarFactory`

**Review by Keywarn — APPROVED** [review]

**Pull request merged**